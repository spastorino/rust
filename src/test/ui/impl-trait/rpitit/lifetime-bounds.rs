trait Foo {}

impl<'x> Foo for &'x mut i32 {}

trait Bar {
    fn bar<'a>(x: &'a mut i32) -> impl Foo + 'a;
}

fn muh<'b, T: Bar>(y: &'b mut i32) {
    let a = T::bar(y);
    let b = T::bar(y);
    //~^ ERROR: cannot borrow `*y` as mutable more than once at a time
    let c = a;
}

fn main() {}
