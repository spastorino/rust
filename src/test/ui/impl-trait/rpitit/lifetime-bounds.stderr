error[E0499]: cannot borrow `*y` as mutable more than once at a time
  --> $DIR/lifetime-bounds.rs:11:20
   |
LL |     let a = T::bar(y);
   |                    - first mutable borrow occurs here
LL |     let b = T::bar(y);
   |                    ^ second mutable borrow occurs here
LL |
LL |     let c = a;
   |             - first borrow later used here

error: aborting due to previous error

For more information about this error, try `rustc --explain E0499`.
